[CS2113T-T10-4] MindMyMoney

GlendonNotGlen opened the pull request:
MindMyMoney is a Command-Line application that allows users to track their spending

wraineflores commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
There is no overview summary of what the software is all about

tuan0369 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+<br/>
+
+### UI component
+The source code can be found in [Ui.java](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Ui.java)
```
I like how you guys include the link for source code in the DG

wraineflores replied:
It's good that there's a source code hyperlink to the Github website

Musfirahe0556596 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
```
Perhaps there should be an activation bar on the call to the AddCommandInputTest#test(...) method as shown in the screenshot?:
![image](https://user-images.githubusercontent.com/69451253/160956872-c75c4fa6-5c73-44f2-964f-0e1b611570e8.png)

ibrahimisramos replied:
the return values should include the data type e.g. category: String

ibrahimisramos replied:
This applies to all the return value

mathanmahe replied:
Would it be possible to enter the objectName:Datatype format for the return types?

Musfirahe0556596 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+   `MindMyMoney`.
+2. `AddCommand` instantiates `addInput`, `expenditureList` and `creditCardList`.
+3. The application invokes `Addcommand.execute` to execute user instruction.
+4. During the execution, `Addcommand.execute` will parse through user input to obtain the `EXPENDITURE`, `CATEGORY`,
+   `DESCRIPTION`, `AMOUNT` and `TIME` fields.
+5. The `Addcommand` object instantiates a new `Expenditure` object with the aforementioned 5 fields and adds them
```
Perhaps there should be another line to describe what the AddCommandInputTest#test(...) method does?

wraineflores commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+![parser_diagram](images/parser_diagram.png)
+<br/> Fig 4 - Parser Diagram
+
+The Parser component consists of a`Parser`, `Functions`, `ExpenditureList` and `Expenditure`class
```
The labeling of figures may be better if placed above the diagram itself. 
Why is the Function class separate from the Parser Class?

tuan0369 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![architecture diagram](images/architecture_diagram.png)
```
The architecture diagram does not have to connection between Parser and Commands, while the command diagram indicates otherwise.
<img width="365" alt="image" src="https://user-images.githubusercontent.com/69478335/160958275-e4e1b0e9-f0b4-4255-a512-f5fa1a8a0563.png">
<img width="735" alt="image" src="https://user-images.githubusercontent.com/69478335/160958342-e56b8e78-1b49-47a2-be01-1fd2d9ecf85d.png">

wraineflores commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+![list_command_sequence_diagram](images/List_Command_Sequence_Diagram.png)
 
 ## Product scope
 ### Target user profile
```
Target User is unknown so it is unclear who this software is for

tuan0369 replied:
There is no information in the product scope, non-functional requirement, glossary and instruction for manual testing sections.
<img width="342" alt="image" src="https://user-images.githubusercontent.com/69478335/160959268-d33cad62-8ba2-4113-a03a-93673cfbd242.png">
<img width="607" alt="image" src="https://user-images.githubusercontent.com/69478335/160959293-b5601eaf-c6c1-4c16-9b65-f23427665b81.png">

Musfirahe0556596 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+To enable users to view their finances in a more meaningfully, MindMyMoney does calculations to present financial data
+that is actionable for the users.
+
+![calculate_command_sequence_diagram](images/gif_loading.gif)
```
Unsure if the gif is supposed to be an indication of **'to be updated'** as here it states that the picture is supposed to be a sequence diagram, but no sequence diagram is shown xD.

wraineflores replied:
Calculate command sequence diagram is not loading maybe because it is a .gif instead of a .png image
![Tutorial Week 11 Screenshot 5b](https://user-images.githubusercontent.com/96967555/160959692-da113307-eb99-4845-8076-8a6dcd0ed565.JPG)

tuan0369 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+### Command component 
+The source can be found in [command](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)
```
The activation bar in the sequence diagram for the Command:AddCommand is too long
<img width="735" alt="image" src="https://user-images.githubusercontent.com/69478335/160958541-84a60512-d30c-47b4-afd0-4b7d77fc6531.png">

Musfirahe0556596 replied:
This diagram is understandable as it shows the different commands available but perhaps an improvement that could be done is to add in the abstract method execute() also to show that all command types have this method in their own classes?

tuan0369 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+To enable users to view their finances in a more meaningfully, MindMyMoney does calculations to present financial data
+that is actionable for the users.
+
+![calculate_command_sequence_diagram](images/gif_loading.gif)
```
I do not know why you guys put loading gif here but I think it should be a sequence diagram
<img width="747" alt="image" src="https://user-images.githubusercontent.com/69478335/160958443-a967d2c7-beff-4f21-8fc9-60fe6c979bf2.png">

ibrahimisramos replied:
I thought it was intended but just curious why haha

mathanmahe replied:
Also curious about the loading gif! Is it a picture that was not rendered?

Musfirahe0556596 commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+The Sequence Diagram below shows an example of how the components interact with each other for the scenario 
+where the user issues the command`add shoes 100`.
+
+![sequence_diagram](images/sequence_diagram.png)
```
Excellent diagram that shows a high-level overview of how the application works. Colouring to differentiate the different components are also good!

ibrahimisramos replied:
For the sequence diagram, shouldn't the user go to :MMM with thecommand add shoes before the :Ui?
The sequence feels a bit off here

mathanmahe replied:
I think they are trying to illustrate that the user enters the command through :UI, which is then sent to  :MMM to handle, which will then parse and do the rest of the sequence.

Musfirahe0556596 reviewed (commented):
Great diagrams drawn! Some issues to consider!

wraineflores commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,150 @@
+![sequence_diagram](images/sequence_diagram.png)
+<br/> Fig 2 - Sequence Diagram
+
+The sections below give more details for each component.
```
There is no termination, loop, opt, etc. in the sequence diagram
![Tutorial Week 11 Screenshot 6b](https://user-images.githubusercontent.com/96967555/160959485-89bf9551-8702-4364-8681-0d36947c4856.JPG)

wraineflores reviewed (commented):
DG peer review

ibrahimisramos commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+The source can be found in [`command`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)  
+Fig 5 - Command Class Diagram
```
Might want to centralise the "figure 5". I took a while to find it

ibrahimisramos commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+
+![add_expenditure_command_sequence_diagram](images/Add_Expenditure_Command_Sequence_Diagram.png)  
```
for the print added expenditure, should it be calling the print()/equivilent print method?

FTang21 replied:
<img width="1001" alt="Screen Shot 2022-04-01 at 10 32 39 AM" src="https://user-images.githubusercontent.com/55422893/161183739-6eb3f1de-f859-480f-af46-fb03b234ace4.png">
I believe your `print added expenditure` self call is missing the activation bar. Similarly, in your addCreditCard() sequence diagram you seem to be missing the `print added credit card` activation bar.

ibrahimisramos commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
```
Might want to add more of the user stories, feels like the app has very limited features

FTang21 replied:
Like ibrahimisramos stated, adding more user stories would be good. It would also tell others what features you focus based on the user stories you looked at or plan to look at.

ibrahimisramos reviewed (commented):
Good job team!

mathanmahe commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+The Sequence Diagram below shows an example of how the components interact with each other for the scenario 
+where the user issues the command`add shoes 100`.
+
+![sequence_diagram](images/sequence_diagram.png)
```
Good use of color schemes to make the sequence and architecture diagrams more visually appealing and easy to follow through!

FTang21 replied:
No critique here, just a really nice diagram sequence that details the architecture of your program!

FTang21 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+
+The sequence diagram below shows the interactions of when an `AddCommand` is parsed.  
+
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
```
<img width="412" alt="Screen Shot 2022-04-01 at 10 22 27 AM" src="https://user-images.githubusercontent.com/55422893/161182071-b2bf652a-57da-4189-aa92-dc4f16c5629a.png">
Shouldn't the return arrow be outside of the alt frame?

FTang21 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+### Command component 
+The source can be found in [`command`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)  
```
<img width="977" alt="Screen Shot 2022-04-01 at 10 25 57 AM" src="https://user-images.githubusercontent.com/55422893/161183008-dadfcb9e-f26a-4b58-8ab0-7f88b58e888a.png">
Should this the lines from Parse to the Commands be dashed? I'm not sure if this a dependency as Parser is creating Commands

mathanmahe commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+### Parser component
+The source code can be found in [`Parser.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Parser.java)
+
+![parser_diagram](images/parser_diagram.png)
```
Would it be possible to clarify this diagram? It is a bit confusing because a package named Parser is used. But the code architecture does not have a package named parser. Would it be possible to rename it to a more suitable name like ExpenditureTracker? Because it houses not only parser, but also the other aspects like the ExpenditureList, Functions as well as Parser

mathanmahe commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+**Value proposition**
+Manage finances containing multiple payment methods faster than a typical mouse/GUI driven app.
+
+### User Stories
```
Would it be possible to add more user stories that could cover what kind of users the features are intended for?

FTang21 reviewed (commented):
Overall really good DeveloperGuide! Good job

mathanmahe commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,300 @@
+* Mainstream OS: Windows, Linux, Unix, OS-X  
 
 ## Instructions for manual testing
 
```
Kindly do add instructions for manual testing of the features so a new developer can understand how to test these features. You can include positive and negative test cases, expected outcomes.

mathanmahe reviewed (approved):
Looks good. It's a very informative developer guide with a few areas for improvement,

khseah committed:
Merge pull request #167 from khseah/kitHan-PPP

Add Kit Han PPP

khseah55 committed:
Update DG

khseah committed:
Merge pull request #168 from khseah/kitHan-PPP

Update PPP

khseah55 committed:
Update AboutUs Portfolio link

khseah committed:
Merge pull request #169 from khseah/kitHan-PPP

Update AboutUs Portfolio link

SeanHoWB committed:
del cmd

SeanHoWB committed:
update cmd

SeanHoWB committed:
Merge pull request #170 from SeanHoWB/Sean-Tests

Sean tests

SeanHoWB committed:
Sean-PPP

SeanHoWB committed:
Merge pull request #171 from SeanHoWB/Sean-PPP

Sean-PPP

SeanHoWB committed:
Sean-PPP fixes

SeanHoWB committed:
Sean-PPP fixes

SeanHoWB committed:
Merge pull request #172 from SeanHoWB/Sean-PPP

Sean ppp

SeanHoWB committed:
Sean-PPP pic changes

SeanHoWB committed:
Merge pull request #173 from SeanHoWB/Sean-PPP

Sean-PPP pic changes

SeanHoWB committed:
DG edits

SeanHoWB committed:
DG edits

SeanHoWB committed:
Merge pull request #174 from SeanHoWB/Sean-DG_edits

DG edits

Commit:
Add Storage and Command class diagrams

Add CalculateInputCommand sequence diagrams

Refactor and shift helper methods to helper class

Commit:
Merge branch 'master' into Branch-Add_DG

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/puml/AddExpenditureCommandSequenceDiagram.puml
#	src/main/java/seedu/mindmymoney/command/AddCommand.java

Commit:
Fix coding standards

SeanHoWB committed:
formatted the print to screens to 2dp

SeanHoWB committed:
fixes

Commit:
Fix comments

Fix add command where invalid input crashes program

Commit:
remove unecessary imports

limjierui committed:
Merge pull request #175 from limjierui/Branch-Add_DG

Add DG

SeanHoWB committed:
fixes pt 2

SeanHoWB committed:
Merge pull request #182 from SeanHoWB/Sean-Bug-Fixes

formatted the print to screens to 2dp

khseah55 committed:
Remove redundant images

khseah55 committed:
Update .gitignore to ignore data.txt

khseah55 committed:
Update AboutUs

khseah55 committed:
Add List Command portion of DG

khseah55 committed:
Update current UML diagrams to have background

khseah committed:
Merge pull request #190 from khseah/kitHan-ListPortionOfDG

Add List portion of DG

khseah55 committed:
Update DG content page

khseah committed:
Merge pull request #191 from khseah/kitHan-ListPortionOfDG

Update DG content page

khseah55 committed:
Update List Command design considerations

khseah committed:
Merge pull request #192 from khseah/kitHan-ListPortionOfDG

Update List Command design considerations

Commit:
Add limjierui PPP

limjierui committed:
Merge pull request #193 from limjierui/Branch-AddPPP

Add limjierui PPP

GlendonNotGlen committed:
Update AddCommandSequence, AddExpenditureCommandSequenceDiagram, CommandClassDiagram, ComponentsSequenceDiagram

GlendonNotGlen committed:
Merge pull request #194 from GlendonNotGlen/master

Update AddCommandSequence, AddExpenditureCommandSequenceDiagram, Comm…

GlendonNotGlen committed:
Remove optional parameters and return values for neater sequence diagrams

GlendonNotGlen committed:
Remove activation bar for calculate command sequence diagram

GlendonNotGlen committed:
Update sequence diagram to conform to UML standards

GlendonNotGlen committed:
Change to add credit card sequence diagram

SeanHoWB committed:
refactor checks to addcommandinputttest

SeanHoWB committed:
refactor checks to addcommandinputttest remove ugly files

SeanHoWB committed:
fixes

SeanHoWB committed:
fixes

SeanHoWB committed:
Merge pull request #196 from SeanHoWB/Sean-Bug-Fixes

refactor checks to addcommandinputttest

GlendonNotGlen committed:
Update sequence diagram to match refactored code

GlendonNotGlen committed:
Merge branch 'master' into master

GlendonNotGlen committed:
Merge pull request #1 from AY2122S2-CS2113T-T10-4/master

Merge conflict

GlendonNotGlen committed:
Update add credit card and add income sequence diagrams

GlendonNotGlen committed:
Merge pull request #197 from GlendonNotGlen/master

Update sequence diagrams

GlendonNotGlen committed:
Add Delete Command section into DG

GlendonNotGlen committed:
Merge pull request #198 from GlendonNotGlen/master

Add Delete Command section into DG

GlendonNotGlen committed:
Update Sequence Diagrams

GlendonNotGlen committed:
Update add command sequence diagram

GlendonNotGlen committed:
Refactor Sequence diagram code

GlendonNotGlen committed:
Merge pull request #199 from GlendonNotGlen/master

Bug Fix sequence diagrams

Commit:
Enhance list command in UG.

Fix list command to be able to take in a date.

Commit:
Remove commented out code

Commit:
Fix coding standard.

Commit:
Fix comments

Commit:
Fix minor date input

limjierui committed:
Merge pull request #200 from limjierui/Branch-Improve_UG

Enhance UG and fix bugs

SeanHoWB committed:
check if cc name in list

SeanHoWB committed:
check if cc name in list

khseah55 committed:
Refine UG

SeanHoWB committed:
Update data.txt

GlendonNotGlen committed:
Update DG with Update Command and update Update Expenditure command

GlendonNotGlen committed:
Update code for coding standards

GlendonNotGlen committed:
Updated Content Table for Update Command

GlendonNotGlen committed:
Update docs/DeveloperGuide.md

Co-authored-by: Sean Ho <69450377+SeanHoWB@users.noreply.github.com>

GlendonNotGlen committed:
Update docs/DeveloperGuide.md

Co-authored-by: Sean Ho <69450377+SeanHoWB@users.noreply.github.com>

GlendonNotGlen committed:
Merge pull request #202 from GlendonNotGlen/branch-Glendon-DG_UpdateCommand

Update DG with Update Command and update Update Expenditure command

SeanHoWB committed:
Update AddCommandInputTests.java

SeanHoWB committed:
Merge pull request #201 from SeanHoWB/Sean-Bug-Fixes

check if cc name in list

SeanHoWB committed:
data

SeanHoWB committed:
Merge branch 'master' of https://github.com/SeanHoWB/tp

Commit:
Add load/store descriptions in DG

danbaterisna committed:
Add links to load/store in DG content page

danbaterisna committed:
Merge branch 'master' into add-load-store-DG

Commit:
Update load/store diagrams

Add colors to activation bars, and use the IntelliJ PlantUML plugin
to generate diagrams

Commit:
Merge branch 'add-load-store-DG' of github.com-danbaterisna:danbaterisna/tp into add-load-store-DG

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/DeserializeListSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/SerializeListSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/SavingSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/LoadingSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/DeserializeListSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/LoadingSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/SavingSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/DeveloperGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Update docs/puml/SerializeListSequenceDiagram.puml

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

limjierui committed:
Merge pull request #203 from danbaterisna/add-load-store-DG

Add load/store in DG

Commit:
Add initial instructions for manual testing

SeanHoWB committed:
fixed credit card limit bugs

Commit:
Merge branch 'master' into add-testing-instructions

khseah55 committed:
Update ListExpenditure portion of DG

SeanHoWB committed:
fixed credit card limit UG

khseah55 committed:
Merge branch 'master' into kitHan-UGTouchUp

# Conflicts:
#	data.txt

khseah committed:
Merge pull request #207 from khseah/kitHan-UGTouchUp

UG touch up + Update ListExpenditure in DG

SeanHoWB committed:
fixed credit card limit UG

GlendonNotGlen committed:
Proofread DG

GlendonNotGlen committed:
Merge pull request #208 from GlendonNotGlen/branch-Glendon-DG_Update

Proofread DG

GlendonNotGlen committed:
Update sequence diagrams

GlendonNotGlen committed:
Merge pull request #210 from GlendonNotGlen/branch-Glendon-DG_Update

Update sequence diagrams

GlendonNotGlen committed:
Refactor code and bug fix sequence diagrams

GlendonNotGlen committed:
Merge pull request #211 from GlendonNotGlen/master

Refactor code and bug fix sequence diagrams

SeanHoWB committed:
ppp rename

Commit:
Add more instructions for manual testing

SeanHoWB committed:
fixes to trailing whitespaces

SeanHoWB committed:
merging conf

Commit:
Standardise border and arrow colour for diagrams in DG

Commit:
Merge branch 'master' into Branch-Fix_DG

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/images/AddCommandSequenceDiagram.png
#	docs/images/DeleteIncomeSequenceDiagram.png

limjierui committed:
Merge pull request #214 from limjierui/Branch-Fix_DG

Standardise border and arrow colour for diagrams in DG

Commit:
Add description of save file format to UG

GlendonNotGlen committed:
Update Glendon PPP

GlendonNotGlen committed:
Merge branch 'master' into branch-Update-PPP

GlendonNotGlen committed:
Merge pull request #2 from AY2122S2-CS2113T-T10-4/branch-Update-PPP

Fix merge conflicts

GlendonNotGlen committed:
Merge pull request #215 from AY2122S2-CS2113T-T10-4/branch-Update-PPP

Update PPP

Commit:
Resolve comments on testing instructions

Specifies the credit limit in Adding 4.ii, separates the section for deleting data,
and adds instructions for testing `list /e {DATE}`.

Commit:
Fix DG review comments

Edit PPP

SeanHoWB committed:
fixed cashback too many dp

SeanHoWB committed:
fixed cashback too many dp fixes

limjierui committed:
Merge pull request #216 from limjierui/Branch-Edit_PPP

Fix DG review comments and edit PPP

SeanHoWB committed:
Update docs/DeveloperGuide.md

Co-authored-by: limjierui <69380595+limjierui@users.noreply.github.com>

SeanHoWB committed:
Merge branch 'master' into Sean-Bug-Fixes

SeanHoWB committed:
format fixes

SeanHoWB committed:
Merge branch 'Sean-Bug-Fixes' of https://github.com/SeanHoWB/tp

GlendonNotGlen committed:
Merge branch 'AY2122S2-CS2113T-T10-4:master' into master

GlendonNotGlen committed:
Update Glendon PPP

SeanHoWB committed:
Merge pull request #206 from SeanHoWB/Sean-Bug-Fixes

Sean bug fixes

GlendonNotGlen committed:
Merge pull request #217 from GlendonNotGlen/master

Small update

Commit:
Add Dan PPP

Commit:
Fix PPP to be within 2 page limit

limjierui committed:
Merge pull request #218 from limjierui/Branch-FixPPPToBeWithinLimit

Fix PPP to be within 2 page limit

danbaterisna committed:
Merge branch 'master' into dan-ug-ppp

khseah55 committed:
Trim PPP to conform with page limit

khseah55 committed:
Merge branch 'master' into kitHan-TrimPPP

SeanHoWB committed:
shorten PPP Sean

khseah55 committed:
Trim PPP further

SeanHoWB committed:
Merge pull request #221 from SeanHoWB/Sean-PPP

shorten PPP Sean

SeanHoWB committed:
Merge pull request #219 from danbaterisna/dan-ug-ppp

Add storage description to UG

khseah55 committed:
Merge branch 'master' into kitHan-TrimPPP

khseah committed:
Merge pull request #220 from khseah/kitHan-TrimPPP

Trim PPP to conform with page limit

SeanHoWB committed:
shorten PPP Sean furhter

Commit:
Cut PPP word count

limjierui committed:
Merge pull request #223 from limjierui/Branch-FurtherCutPPP

Cut PPP word count

SeanHoWB committed:
Merge pull request #222 from SeanHoWB/Sean-PPP

shorten PPP Sean furhter

SeanHoWB committed:
shorten PPP Sean furhterer

SeanHoWB committed:
Merge pull request #224 from SeanHoWB/Sean-PPP

shorten PPP Sean furhterer

SeanHoWB committed:
shorten PPP Sean furhterer

SeanHoWB committed:
Merge pull request #225 from SeanHoWB/Sean-PPP

shorten PPP Sean furhterer

Commit:
Cut even more words in PPP

limjierui committed:
Merge pull request #226 from limjierui/Branch-FutherCutPPP2

Cut even more words in PPP

khseah55 committed:
Trim PPP further

Commit:
Final cut PPP

SeanHoWB committed:
coming soon feature

limjierui committed:
Merge pull request #228 from limjierui/Branch-FinalCutPPP

Final cut PPP

khseah55 committed:
Trim PPP for the last time

khseah55 committed:
Merge branch 'master' into kitHan-TrimPPPFurther

SeanHoWB committed:
Merge pull request #229 from SeanHoWB/Sean-UG-fixes

coming soon feature

khseah55 committed:
Merge branch 'master' into kitHan-TrimPPPFurther

khseah committed:
Merge pull request #227 from khseah/kitHan-TrimPPPFurther

Trim PPP further

SeanHoWB committed:
changed app to application

GlendonNotGlen committed:
Update Update Credit Card command for duplicate card

GlendonNotGlen committed:
Merge branch 'master' into branch-Glendon-FixUpdateCC

# Conflicts:
#	src/main/java/seedu/mindmymoney/command/UpdateCommand.java

GlendonNotGlen committed:
Update for coding standards

GlendonNotGlen committed:
Update JUnit testing for update CC command

GlendonNotGlen committed:
Merge pull request #231 from GlendonNotGlen/branch-Glendon-FixUpdateCC

Branch glendon fix update cc

SeanHoWB committed:
intro to Commands

Commit:
Add validators for reading save file

Fixes #212

SeanHoWB committed:
more formatting

SeanHoWB committed:
more formatting of content page

Commit:
Update About Us with link to Dan PPP

SeanHoWB committed:
KH ask me to superscript

SeanHoWB committed:
cont page intro to commands

danbaterisna committed:
Merge pull request #213 from danbaterisna/add-testing-instructions

Add testing instructions

SeanHoWB committed:
null input credit card fix

SeanHoWB committed:
Update docs/UserGuide.md

Co-authored-by: khseah <69412944+khseah@users.noreply.github.com>

SeanHoWB committed:
Merge pull request #230 from SeanHoWB/Sean-UG-fixes

changed app to application

SeanHoWB committed:
null input credit card fix fixes

SeanHoWB committed:
Merge pull request #233 from SeanHoWB/Sean-Bugs-3hrs-to-deadline

Sean bugs 3hrs to deadline

Commit:
Correct spacing

Commit:
Standardise sequence and class diag format

khseah55 committed:
Finalize UG

GlendonNotGlen committed:
Update sequence diagram

khseah committed:
Merge pull request #235 from khseah/kitHan-FinalizeUG

Finalize UG

GlendonNotGlen committed:
Merge pull request #236 from GlendonNotGlen/master

Update sequence diagram

khseah55 committed:
Shift calculate cashback to under CreditCard

Commit:
Fix colour of diagram

khseah committed:
Merge pull request #237 from khseah/kitHan-FinalizeUG

Shift calculate cashback to under CreditCard

GlendonNotGlen committed:
Minor edit to UG

GlendonNotGlen committed:
Merge branch 'master' into master

SeanHoWB committed:
fix line

SeanHoWB committed:
fix lines

SeanHoWB committed:
fix lines

SeanHoWB committed:
fix lines again

Commit:
Fix diagrams

Commit:
Merge branch 'master' into Branch-StandardiseDGDiag

# Conflicts:
#	docs/images/AddCreditCardSequenceDiagram.png
#	docs/images/AddExpenditureSequenceDiagram.png

SeanHoWB committed:
fix some bugs

limjierui committed:
Merge pull request #234 from limjierui/Branch-StandardiseDGDiag

Standardise sequence and class diag format

SeanHoWB committed:
Merge pull request #238 from SeanHoWB/Sean-Bugs-1hr-to-deadline

fix line

SeanHoWB committed:
fix regex

SeanHoWB committed:
Merge pull request #239 from SeanHoWB/Sean-Bugs-lessthan-1hr-to-deadline

fix regex

SeanHoWB committed:
fix 1mil

SeanHoWB committed:
fixed ug to match 1 mil

khseah55 committed:
Add max_income limit

khseah55 committed:
Merge branch 'master' into kitHan-setMaxIncomeAmount

danbaterisna committed:
Merge branch 'master' into validate-saved-info

SeanHoWB committed:
fixed ug to match 1 mil

Commit:
Migrate PropertyList to data package

SeanHoWB committed:
Merge pull request #240 from SeanHoWB/Sean-Bugs-lessthan-1hr-to-deadline

Sean bugs lessthan 1hr to deadline

SeanHoWB committed:
fixed bug of startwith

SeanHoWB committed:
Merge pull request #232 from danbaterisna/validate-saved-info

Add validators for reading save file

SeanHoWB committed:
Merge pull request #242 from SeanHoWB/Sean-Bugs-lessthan-1hr-to-deadline

fixed bug of startwith

khseah55 committed:
Check for invalid income amount

SeanHoWB committed:
change dan face

khseah55 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-4/tp into kitHan-setMaxIncomeAmount

SeanHoWB committed:
change dan face

SeanHoWB committed:
Merge pull request #243 from SeanHoWB/Sean-Bugs-lessthan-1hr-to-deadline

change dan face

khseah committed:
Merge pull request #241 from khseah/kitHan-setMaxIncomeAmount

Kit han set max income amount

SeanHoWB committed:
change dan face pt 2

SeanHoWB committed:
Merge pull request #244 from SeanHoWB/Sean-help_me

change dan face pt 2

Commit:
Minor changes to command classes

limjierui committed:
Merge pull request #245 from limjierui/Branch-LastChanges

Minor changes to command classes

GlendonNotGlen committed:
Rename GlendonNotGlen.md to glendonnotglen.md

GlendonNotGlen committed:
Update AboutUs.md
